Source/VMware.vSphereDSC/DSCResources/DatastoreCluster.ps1

```diff
@@ -0,0 +1,299 @@
+            $result = $null
+
+            if ($this.Ensure -eq [Ensure]::Present) {
+                if ($null -eq $datastoreCluster) {
+                    $result = $false
+                }
+                else {
+                    $result = !$this.ShouldModifyDatastoreCluster($datastoreCluster)
+                }
+            }
+            else {
+                $result = ($null -eq $datastoreCluster)
+            }
+
+            $this.WriteDscResourceState($result)
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.TestMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    [DatastoreCluster] Get() {
+        try {
+            Write-VerboseLog -Message $this.GetMethodStartMessage -Arguments @($this.DscResourceName)
+            $result = [DatastoreCluster]::new()
+
+            $this.ConnectVIServer()
+
+            $datacenter = $this.GetDatacenter()
+            $datacenterFolderName = "$($this.InventoryItemFolderType)Folder"
+            $datastoreClusterLocation = $this.GetInventoryItemLocationInDatacenter($datacenter, $datacenterFolderName)
+
+            $datastoreCluster = $this.GetDatastoreCluster($datastoreClusterLocation)
+            $this.PopulateResult($result, $datastoreCluster)
+
+            return $result
+        }
+        finally {
+            $this.DisconnectVIServer()
+            Write-VerboseLog -Message $this.GetMethodEndMessage -Arguments @($this.DscResourceName)
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Datastore Cluster with the specified name located in the specified Folder if it exists.
+    #>
+    [PSObject] GetDatastoreCluster($datastoreClusterLocation) {
+        $getDatastoreClusterParams = @{
+            Server = $this.Connection
+            Name = $this.Name
+            Location = $datastoreClusterLocation
+            ErrorAction = 'SilentlyContinue'
+            Verbose = $false
+        }
+
+        $whereObjectParams = @{
+            FilterScript = {
+                $_.ExtensionData.Parent -eq $datastoreClusterLocation.ExtensionData.MoRef
+            }
+        }
+
+        <#
+            Multiple Datastore Clusters with the same name can be present in a Datacenter. So we need to filter
+            by the direct Parent Folder of the Datastore Cluster to retrieve the desired one.
+        #>
+        return Get-DatastoreCluster @getDatastoreClusterParams | Where-Object @whereObjectParams
+    }
+
+    <#
+    .DESCRIPTION
+
+    Checks if the specified Datastore Cluster configuration should be modified.
+    #>
+    [bool] ShouldModifyDatastoreCluster($datastoreCluster) {
+        $shouldModifyDatastoreCluster = @()
+
+        $shouldModifyDatastoreCluster += ($null -ne $this.IOLatencyThresholdMillisecond -and $this.IOLatencyThresholdMillisecond -ne $datastoreCluster.IOLatencyThresholdMillisecond)
```

Comment:
May I suggest using a method to do all these compares? That way you could also tackle the issue described in #264, it also prevents a lot of code duplication. For example that method could look like: 

```PowerShell
[bool] CompareProperty($unsetValue, $intentValue, $currentValue, $propertyName){
    if ($unsetValue -ne $intentValue -and $intentValue -ne $currentValue) {
        Write-Verbose "Setting $propertyName: Current setting [ $currentValue ] does not match desired setting [ $intentValue ]"
        return $false
    } else {
        return $true
    }
}
```

calling it in this function would then be:
```PowerShell
[bool] ShouldModifyDatastoreCluster($datastoreCluster) {
    $shouldModifyDatastoreCluster = @(
        $this.CompareProperty($null, $this.IOLatencyThresholdMillisecond, $datastoreCluster.IOLatencyThresholdMillisecond, 'IOLatencyThresholdMillisecond')
        $this.CompareProperty($null, $this.IOLoadBalanceEnabled, $datastoreCluster.IOLoadBalanceEnabled, 'IOLoadBalanceEnabled')
        $this.CompareProperty([DrsAutomationLevel]::Unset, $this.SdrsAutomationLevel.ToString(), $datastoreCluster.SdrsAutomationLevel.ToString(), 'SdrsAutomationLevel')
        $this.CompareProperty($null, $this.SpaceUtilizationThresholdPercent, $datastoreCluster.SpaceUtilizationThresholdPercent, 'SpaceUtilizationThresholdPercent')
    )

    return ($shouldModifyDatastoreCluster -contains $false)
}
```

Reply:
Great addition! We can add it to the **BaseDSC** class and then call it in each DSC Resource in the module. I will start working on that now.